### PR TITLE
Adding additional .mmi extensions to all possible fasta file format l…

### PR DESCRIPTION
**Pull request by mattloose**
…abels to catch acceptable file types.

This request addresses #326 and enables validation of the following file type endings.

['.fasta.gz', '.fna.gz', '.fsa.gz', '.fa.gz', '.fastq.gz', '.fq.gz', '.fasta.mmi', '.fna.mmi', '.fsa.mmi', '.fa.mmi', '.fastq.mmi', '.fq.mmi', '.fasta', '.fna', '.fsa', '.fa', '.fastq', '.fq', '.mmi']

I think a better way to handle this in the future would be to explicitly detemine if the file is readable by mappy. However, this will catch the most common possible mmi extension types.

### CODE REVIEW

**Comment by alexomics** [comment]
I don't think that the current fix addresses the problem in the issue. Which is an extra `.` in the filename (`GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.mmi`), so pathlib suggests that the suffix is `.15_grch38_no_alt_analysis_set.fna.mmi` 

We could use something like:
```python
file_extensions = ['.fasta', '.fna', '.fsa', '.fa', '.fastq', 
                   '.fq', '.fasta.gz', '.fna.gz', '.fsa.gz', 
                   '.fa.gz', '.fastq.gz', '.fq.gz', '.mmi']
if not any(index.lower().endswith(suffix) for suffix in file_extensions):
    raise ...
```

Probably should add a test case TOML for this too.

**Comment by mattloose** [comment]
I mean - it does fix it...

But I'll look at this alternative.

**Comment by mattloose** [comment]
OK
It doesn't fix it.

Whoops.

File: src/readfish/plugins/_mappy.py [hunk 1]
```diff
@@ -82,15 +82,12 @@ def validate(self) -> None:
         :return: None, if the Aligner is setup with valid paths and permissions
         """
         index: str = self.aligner_params["fn_idx_in"]
-        file_extensions = [".fasta", ".fna", ".fsa", ".fa", ".fastq", ".fq"]
-        file_extensions.extend([f"{f}.gz" for f in file_extensions])
-        file_extensions.append(".mmi")
+        file_extensions = ['.fasta', '.fna', '.fsa', '.fa', '.fastq',
+                           '.fq', '.fasta.gz', '.fna.gz', '.fsa.gz',
+                           '.fa.gz', '.fastq.gz', '.fq.gz', '.mmi']
```
Comment (alexomics) [inline src/readfish/plugins/_mappy.py:87]: I would keep the removed lines here, makes updating this list much easier in the future.

**Review by alexomics — CHANGES_REQUESTED** [review]
I think I would also add another test case that has a valid suffix but strange filename e.g. `test.v1.fna.mmi`

**Comment by alexomics** [comment]
So the TOML tests are a bit of a Rube Goldberg machine, you need to:
1. create a TOML file (`readfish/tests/static/toml_validation_test/fail/005_bad_reference_file.TOML`)
1. create a matching TXT file (`readfish/tests/static/toml_validation_test/fail/005_bad_reference_file.txt`) that contains the expected error message(s) the TOML will generate

It's described a little bit more on the [README](https://github.com/LooseLab/readfish/blob/3fb5df79d5149b67ee204f65809b58794fbfc8ff/tests/static/mappy_validation_test/README.md) for each test folder.

File: src/readfish/plugins/_mappy.py (hunk 1 above)
Comment (mattloose) [reply src/readfish/plugins/_mappy.py]: Done

**Comment by mattloose** [comment]
re: the tests - the output of the readfish validate command won't error on this file now - it will just give a warning - can that be caught in a text file? I note that in the other examples for validating mappy the text file is just empty?

**Comment by mattloose** [comment]
OK - this adds a fail and pass test for these specific issues - the pass reference index file is named as per the issue that started this.

**Review by alexomics — APPROVED** [review]

**Pull request merged**